### PR TITLE
Infra/setup bastion host for new cluster rds

### PR DESCRIPTION
**Pull request by ce0la**
## Summary by CodeRabbit

- **New Features**
  - Implemented a search functionality with a search bar and display components.

- **Style**
  - Added styles for the search bar.

### CODE REVIEW

File: infra/us-east-1/prod/rds/bastion/security-group/terragrunt.hcl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+dependency "vpc" {
+  config_path = "../../../vpc"
+}
+
+inputs = {
+  vpc_id = dependency.vpc.outputs.vpc_id
+
+  computed_ingress_cidr_blocks = ["0.0.0.0/0"]
+  ingress_cidr_blocks          = ["0.0.0.0/0"]
```
Comment (renebarbosafl) [inline infra/us-east-1/prod/rds/bastion/security-group/terragrunt.hcl:32]: As you're already setting VPC as a dependency, why not using the outputs to declare these CIDRs?

```hcl
inputs = {
  name                = local.security_group_name
  vpc_id              = dependency.vpc.outputs.vpc_id
  ingress_cidr_blocks = dependency.vpc.outputs.private_subnets_cidr_blocks
}
```

File: infra/us-east-1/prod/rds/bastion/security-group/terragrunt.hcl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  app            = "marlowe-runtime"
+
+  bastion_name          = "${local.project}-${local.app}-bastion"
+  bastion_key_pair_name = "${local.project}-${local.app}-db-bastion-kp"
+
+  tags = {
+    Owner        = "Terraform"
+    tribe        = "smart-contracts"
+    App          = "marlowe-runtime"
+  }
```
Comment (renebarbosafl) [inline infra/us-east-1/prod/rds/bastion/security-group/terragrunt.hcl]: Default tags are already set up on the provider (using default_tags parameter). Probably you don't need to declare these.

File: infra/us-east-1/prod/rds/looker-studio-db/main.tf
Comment (renebarbosafl) [inline infra/us-east-1/prod/rds/looker-studio-db/main.tf:1]: Why not using Terragrunt for this resource to keep the repo pattern?
Example: https://github.com/input-output-hk/sc-dev-platform/blob/main/infra/us-east-1/prod/rds/prod-marlowe-runtime-db/terragrunt.hcl

File: infra/us-east-1/prod/rds/bastion/key-pair/terragrunt.hcl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    organization = "iog"
+    tribe        = "smart-contracts"
+    environment  = "prod"
+    function     = "bastion-key-pair"
+    App          = "marlowe-runtime"
+    Environment  = "prod"
+    Terraform    = "true"
+    Project      = local.project
+    Resource     = local.bastion_name
+  }
```
Comment (renebarbosafl) [inline infra/us-east-1/prod/rds/bastion/key-pair/terragrunt.hcl]: Please check if we can use the tags defined by the provider: 
https://github.com/input-output-hk/sc-dev-platform/blob/main/infra/terragrunt.hcl#L64

File: infra/us-east-1/prod/rds/bastion/ec2-instance/terragrunt.hcl [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  tags = {
+    organization = "iog"
+    tribe        = "smart-contracts"
+    environment  = "prod"
+    function     = "bastion"
+    Terraform    = "true"
+    Project      = local.project
+    Resource     = local.bastion_name
+  }
```
Comment (renebarbosafl) [inline infra/us-east-1/prod/rds/bastion/ec2-instance/terragrunt.hcl]: Please check if we can use the tags defined by the provider: 
https://github.com/input-output-hk/sc-dev-platform/blob/main/infra/terragrunt.hcl#L64

File: infra/us-east-1/prod/rds/bastion/ec2-instance/terragrunt.hcl [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  # Automatically load environment-level variables
+  account_vars     = read_terragrunt_config(find_in_parent_folders("account.hcl"))
+
+  # Extract out common variables for reuse
+  project        = local.account_vars.locals.project
+  app            = "marlowe-runtime"
+
+  bastion_name       = "${local.project}-${local.app}-database-bastion"
+
+  subnet_id              = "subnet-087617570cd189b3e"
```
Comment (renebarbosafl) [inline infra/us-east-1/prod/rds/bastion/ec2-instance/terragrunt.hcl]: Probably it's a good idea to fetch this from a dependency instead of hardcoding.

**Review by renebarbosafl — CHANGES_REQUESTED** [review]

File: infra/us-east-1/prod/rds/bastion/ec2-instance/terragrunt.hcl (hunk 5 above)
Comment (ce0la) [reply infra/us-east-1/prod/rds/bastion/ec2-instance/terragrunt.hcl]: Yes, it is being fetched from a dependency, see here: https://github.com/input-output-hk/sc-dev-platform/blob/43cf87dd2ee12ac0d1fdb7ef05c49604989cd7ce/infra/us-east-1/prod/rds/bastion/ec2-instance/terragrunt.hcl#L56. 

Forgot to take this local variable out.

File: infra/us-east-1/prod/rds/bastion/security-group/terragrunt.hcl (hunk 1 above)
Comment (ce0la) [reply infra/us-east-1/prod/rds/bastion/security-group/terragrunt.hcl:32]: These are supposed to be the CIDRs for allowed IPs in the security group. Those can't be extracted from the VPC output.

File: infra/us-east-1/prod/rds/looker-studio-db/main.tf
Comment (ce0la) [reply infra/us-east-1/prod/rds/looker-studio-db/main.tf:1]: This is just a remote store for the file that is used in the cron job on the Bastion host. There was no need to set up Terragrunt on the host (an unneeded layer of abstraction, tbh), so I decided to go with a simple TF file which I have committed here.

Comment (renebarbosafl) [reply infra/us-east-1/prod/rds/looker-studio-db/main.tf:1]: @ce0la Thank you!

Could you please add some doc about how the process of replicating this database works? 
I believe the main informations we need to have are:

1. Where is the cronjob running?
2. What's the current schedule?
3. Are there some known issues?

A markdown file in this repo should work.

File: infra/us-east-1/prod/rds/looker-studio-db/NOTES.md
Comment (renebarbosafl) [inline infra/us-east-1/prod/rds/looker-studio-db/NOTES.md:1]: @ce0la , Could you please add some doc about how the process of replicating this database works?
I believe the main information we need is:

1. Where is the cronjob running?
2. What's the current schedule?
3. Are there some known issues? If so, please elaborate.

**Review by renebarbosafl — CHANGES_REQUESTED** [review]

**Comment by ce0la** [comment]
Became a horror-show with the rebases/updates, so opened a new PR instead: https://github.com/input-output-hk/sc-dev-platform/pull/67

Closing this.

**Pull request closed without merging**